compiler/noirc_evaluator/src/ssa/ir/dfg.rs

```diff
@@ -223,11 +224,23 @@ impl DataFlowGraph {
         instruction_data: Instruction,
         ctrl_typevars: Option<Vec<Type>>,
     ) -> InstructionId {
+        self.validate_instruction(&instruction_data);
+
         let id = self.instructions.insert(instruction_data);
         self.make_instruction_results(id, ctrl_typevars);
         id
     }
 
+    fn validate_instruction(&self, instruction: &Instruction) {
+        if let Instruction::Binary(Binary { lhs, rhs: _, operator: BinaryOp::Lt }) = instruction {
+            // Assume rhs_type is the same as lhs_type
+            let lhs_type = self.type_of_value(*lhs);
+            if matches!(lhs_type, Type::Numeric(NumericType::NativeField)) {
+                panic!("Cannot use `lt` with field elements");
+            }
```

Comment:
Could https://github.com/noir-lang/noir/blob/354c5aa6a432d28c0d62be6fc8114f1add09f44e/compiler/noirc_frontend/src/ast/expression.rs#L332 be of use?